frontend/packages/knative-plugin/src/utils/create-eventsources-utils.ts

```diff
@@ -26,15 +27,37 @@ export const getEventSourcesDepResource = (formData: any): K8sResourceKind => {
     project: { name: namespace },
     data,
     sink: { knativeService },
+    limits: { cpu, memory },
   } = formData;
 
   const defaultLabel = getAppLabels(name, applicationName);
   const apiGroup =
-    type === EventSources.ApiServerSource || type === EventSources.SinkBinding
+    type === EventSources.ApiServerSource ||
+    type === EventSources.SinkBinding ||
+    type === EventSources.KafkaSource
       ? KNATIVE_EVENT_SOURCE_APIGROUP
       : KNATIVE_EVENT_SOURCE_APIGROUP_DEP;
   const apiVersion = 'v1alpha1';
   const eventSrcData = data[type.toLowerCase()];
+  const completeEventSrcData =
+    type === EventSources.KafkaSource && eventSrcData
```

Comment:
am not sure if we can avoid this but wanted to have this utils without anyModel check. Working on eliminating above for apiGroup as well.

Reply:
I have refactored the utils and now I have created a base util that returns the common/base resource and event source specific utils to modify the extend the base resource as needed. https://github.com/openshift/console/pull/4899/commits/23149f5c30fad6039f6e491ffb1f7494e23f0c26.